x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyBoolQueryBuilder.java

```diff
@@ -139,13 +140,43 @@ private static QueryBuilder doProcess(QueryBuilder qb, Consumer<String> fieldNam
         } else if (qb instanceof final PrefixQueryBuilder query) {
             final String translatedFieldName = ApiKeyFieldNameTranslators.translate(query.fieldName());
             fieldNameVisitor.accept(translatedFieldName);
-            return QueryBuilders.prefixQuery(translatedFieldName, query.value()).caseInsensitive(query.caseInsensitive());
+            return QueryBuilders.prefixQuery(translatedFieldName, query.value())
+                .caseInsensitive(query.caseInsensitive())
+                .rewrite(query.rewrite());
```

Comment:
Is this a bug fix? If it is it would be nice to add a test for it.

Reply:
I've added the test in [06252a9](https://github.com/elastic/elasticsearch/pull/104594/commits/06252a99888bf33cac71f6403b0aea3107b32009).